### PR TITLE
#1-Database_schema-Robert

### LINKED ISSUE #1: Task: Create database schema for energisers
energiser: energiser_id, name, description:text

energiser_rating: rating_id, energiser_id, rating:int

### PR DESCRIPTION
**Pull request by RobCso**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

I have created a database schema and set up the local database- this will have to be changed to the local user's settings

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: server/cyf.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
+DROP TABLE IF EXISTS energizers CASCADE;
+DROP TABLE IF EXISTS raitings CASCADE;
```
Comment (bertilnilsson) [inline server/cyf.sql]: Spelling is "ratings", also I suggest using the name `energizer_ratings` since it's a sub-entity to `energizer`

File: server/cyf.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+CREATE TABLE energizers (
+    id           SERIAL PRIMARY KEY,
+    name         VARCHAR(50) NOT NULL,
+    energizer    VARCHAR NOT NULL
```
Comment (bertilnilsson) [inline server/cyf.sql]: I suggest using `description` instead of `energizer`

File: server/cyf.sql [hunk 3]
```diff
@@ -0,0 +1,20 @@
+DROP TABLE IF EXISTS energizers CASCADE;
+DROP TABLE IF EXISTS raitings CASCADE;
+
+CREATE TABLE energizers (
+    id           SERIAL PRIMARY KEY,
+    name         VARCHAR(50) NOT NULL,
```
Comment (bertilnilsson) [inline server/cyf.sql]: Is there any reason this is limited to 50 characters?

File: server/cyf.sql [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    energizer    VARCHAR NOT NULL
+);
+
+CREATE TABLE raitings (
+    id              SERIAL PRIMARY KEY,
+    energizer_id    integer REFERENCES energizers(id),
+    raiting         integer NOT NULL
```
Comment (bertilnilsson) [inline server/cyf.sql]: Spelling

File: server/cyf.sql [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+);
+
+INSERT INTO energizers(name, energizer) VALUES ('hello world',' another jolt that explores one of our favorite themes: You have to unlearn something old in order to learn something new. A nice thing about this brief activity is that you dont need any supplies or equipment.Goal
+To experience problems associated with unlearning previous associations. Instructions'
+);
+
+INSERT INTO raitings(energizer_id,raiting) VALUES (1,5);
```
Comment (bertilnilsson) [inline server/cyf.sql]: This is assuming the that the previous line gets inserted with an id of 1 which is not going to be necessarily true

File: server/server.js [hunk 6]
```diff
@@ -15,4 +18,18 @@ server.on("listening", () => {
...
+app.get("/hello", (req, res) => {
+	res.json({ message: "Hello World" });
+});
+
+// app.post("/", (req, res) => {
+// 	const insertQ = "INSERT INTO energizers(name,energizer) VALUES($1,$2)";
+// 	db.query(insertQ, [req.name, req.energizer]).then(() => {
+// 		res.status(201).json({ message: "table updated" });
+// 	});
+// });
```
Comment (bertilnilsson) [inline server/server.js]: This commented out code should not be part of the pull request

File: server/static/index.html [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
```
Comment (bertilnilsson) [inline server/static/index.html]: Why has this file been added?

File: server/server.js [hunk 8]
```diff
@@ -1,10 +1,13 @@
 import http from "http";
+import { Pool } from "pg";
 
 import app from "./app";
 import { connectDb, disconnectDb } from "./db";
 import config from "./utils/config";
 import logger from "./utils/logger";
 
+//import db from "./db";
```
Comment (bertilnilsson) [inline server/server.js]: Why is this commented out line here?

**Comment by bertilnilsson** [comment]
@RobCso I've create a PR #12 for using knex to handle database migrations (just copying your current schema) that you might want to base of/take inspiration from.

File: server/cyf.sql (hunk 1 above)
Comment (RobCso) [reply server/cyf.sql]: Thank you for spotting this, good suggestion

File: server/cyf.sql (hunk 2 above)
Comment (RobCso) [reply server/cyf.sql]: Good suggestion thank you

File: server/cyf.sql (hunk 3 above)
Comment (RobCso) [reply server/cyf.sql]: The name should be short and descriptive

File: server/cyf.sql (hunk 4 above)
Comment (RobCso) [reply server/cyf.sql]: thank you for spotting this

File: server/cyf.sql (hunk 5 above)
Comment (RobCso) [reply server/cyf.sql]: as my SQL knowledge is limited I have assumed that the drop table if exist command will replace the existing table if exists with a new one, I have followed examples from previous exercises from CYF that populated the database schema with example data,

File: server/server.js (hunk 6 above)
Comment (RobCso) [reply server/server.js]: I have removed this part in my last commit before the pull request

File: server/static/index.html (hunk 7 above)
Comment (RobCso) [reply server/static/index.html]: I have removed this, I have unnecessarily added it while trying to work out some errors I was getting while trying to figure out the template

**Comment by bertilnilsson** [comment]
I think we could merge this, I will close my PR in favour of this.
